### ISSUE TITLE
Redo: apples-to-apples SmolLM2 strict-mix distillation baseline

### ISSUE THREAD

**Issue by tokenbender** [tokenbender/prism-capability-extraction#16]
## Scope
Redo the SmolLM2 dense distillation comparison using the same SFT data boundary and evaluation standard as the Qwen3-8B BFCL adapter path.

Tracked under #1. Supersedes the scientific claim in #15, which trained on the benchmark train split and therefore answered a different question.

## Why
#15 showed that SmolLM2 can learn the local BFCL benchmark train split very strongly, but that was not the apples-to-apples comparison against the Qwen3-8B adapter lineage. The Qwen3-8B path used a separate strict BFCL-compatible 10k SFT mix, leak-audited against the 1007-row BFCL eval surface, then evaluated on the full 1007 rows with the BFCL canonicalization prompt and normalized exact scorer.

The corrected question is:

> If SmolLM2 gets the same strict 10k SFT mix and the same full-BFCL validation/eval standard as the Qwen3-8B adapter path, how much does it improve and does it still beat the PRISM/MACE comparator?

## Corrected Data Boundary
Training data:

```text
data/bfcl_strict_10k_mix_len1024/train.jsonl
```

Known Qwen3-8B adapter-path facts from the handoff / repo receipts:
- rows: `9999`
- leak audit against `data/bfcl_single_call/pairs.jsonl`: exact row overlaps `0`, exact prompt overlaps `0`, exact target-only overlaps `0`, near overlaps at `0.85`: `0`
- old Qwen3-8B adapter path: `Qwen/Qwen3-8B`, LoRA r32/alpha64, max seq len `1024`, one epoch / `1250` optimizer steps in the H200 handoff

Evaluation data:

```text
data/bfcl_single_call/pairs.jsonl
```

Evaluation standard:
- full `1007` rows, not benchmark-train selection
- BFCL canonicalization prompt enabled, matching `bfcl_direct_qwen3.py eval-mask --bfcl-canonicalization-prompt --normalized`
- normalized exact structured scorer
- report category scores and invalid tool-call rate
- split manifest may be reported for diagnostics, but selection must not use benchmark train rows

## Required Correction To #15
#15 remains useful as an overfit/train-split stress test, but its closeout must not be cited as the apples-to-apples distillation answer. This issue is the corrected comparison.

## Implementation Plan
1. Extend the dense student runner to train from strict BFCL-compatible JSONL with `target_text`.
2. Add Qwen-style `chat-tools` prompt rendering so SFT/eval can mirror the old Qwen3 BFCL harness rather than the #15 rendered benchmark prompt.
3. Run an OOB SmolLM2 eval under this corrected prompt/eval standard.
4. Profile one full-load strict-mix SFT configuration before the full run.
5. Train SmolLM2 on the strict 10k mix, preferably full SFT unless utilization or memory makes LoRA the only practical option.
6. Evaluate the trained checkpoint on the full 1007 BFCL eval surface with the corrected standard.
7. Preserve selected model/adapter, train summaries, eval JSONL/summaries, profile receipts, and checksums before pod shutdown.

## Candidate
Primary candidate:

```text
HuggingFaceTB/SmolLM2-1.7B-Instruct
```

Initial corrected plan:
- train mode: full SFT first
- train JSONL: `data/bfcl_strict_10k_mix_len1024/train.jsonl`
- max seq length: `1024`
- prompt style: `chat-tools`
- train canonicalization prompt: off, unless the old strict-mix recipe proves otherwise
- eval canonicalization prompt: on
- eval prompt style: `chat-tools`
- evaluator: vLLM prompt-compatible scorer plus normalized exact; if `chat-tools` is unsupported by SmolLM2 tokenizer, record that as a blocker and use a documented fallback only after noting the mismatch

## Acceptance Criteria
- [ ] #15 is explicitly marked as superseded for the apples-to-apples scientific claim.
- [ ] Dense-student runner supports strict train JSONL and Qwen-style `chat-tools` prompt mode.
- [ ] Corrected OOB SmolLM2 score is measured on all 1007 rows under the Qwen-style eval standard.
- [ ] Leak/data boundary is restated: strict train mix is separate from the 1007 eval rows.
- [ ] Full-load profile gate is posted before full training.
- [ ] Strict-mix SmolLM2 SFT run completes or a blocker is recorded with exact evidence.
- [ ] Full 1007-row post-SFT eval is recorded with normalized exact, invalid calls, and category scores.
- [ ] Result is compared against #12 PRISM/MACE anchors: `600/1007 @ k=140,875` and `607/1007 @ k=141,250`.
- [ ] Artifacts and SHA256 checksums are preserved locally before any pod is terminated.
- [ ] Final label is one of: `distillation wins`, `PRISM wins`, `mixed result`, `invalid comparison`.
